### PR TITLE
Feature/add services

### PR DESCRIPTION
**Pull request by cleprt**

### CODE REVIEW

File: python/ros2asn1/RosAsn1Generator.py [hunk 1]
```diff
@@ -36,30 +36,51 @@ def __init__(self, rospack, pkg_name):
...
         '''List of names of the messages defined by the package'''
         msgs = rosmsg.list_msgs(self.pkg_name, self.rospack)
+        srvs = rosmsg.list_srvs(self.pkg_name, self.rospack)
         # Remove prefix 'package_name/'
         msgs = [msg.replace(self.pkg_name + '/', '', 1) for msg in msgs]
-        return msgs
+        srvs = [srv.replace(self.pkg_name + '/', '', 1) for srv in srvs]
+
+        self.has_msg = ( len(msgs) > 0) 
+        self.has_service = ( len(srvs) > 0)
```
Comment (vbissonnette) [inline python/ros2asn1/RosAsn1Generator.py]:
```suggestion
        self.has_msg = (len(msgs) > 0) 
        self.has_service = (len(srvs) > 0)
```

File: python/ros2asn1/RosAsn1Generator.py [hunk 2]
```diff
@@ -136,7 +157,8 @@ def split_type(self, typename):
     def required_packages(self):
         '''Returns the list of packages needed for all the messages and slots'''
         required = []
-        for msg in self.messages():
+        msgs = self.messages()
+        for msg in msgs:
```
Comment (vbissonnette) [inline python/ros2asn1/RosAsn1Generator.py]: Is this change necessary ?

File: python/ros2asn1/ros2asn1.py [hunk 3]
```diff
@@ -39,17 +39,35 @@ def process_all_messages(out_dir):
...
 
-        # ASN.1 types "pkg.asn"
-        asn_txt = asn_template.render(pkg=pkg_obj)
+    #check that packages given in input exists
+    if packages_to_process:
+        for pkg in packages_to_process:
+            if pkg not in existing_packages:
+                print("process_all_messages : input package {} does not exist".format(pkg))
+                sys.exit(-1)
+                return false
```
Comment (vbissonnette) [inline python/ros2asn1/ros2asn1.py]: `sys.exit(-1)` exits the program entirely. I don't think `return false` is called at all.

**Review by vbissonnette — CHANGES_REQUESTED** [review]

File: python/ros2asn1/RosAsn1Generator.py (hunk 2 above)
Comment (cleprt) [reply python/ros2asn1/RosAsn1Generator.py]: No its not, i remove it

**Review by vbissonnette — APPROVED** [review]

**Pull request merged**